### ISSUE TITLE
robot-simulator: add exercise, exercise

### ISSUE THREAD

**Issue by rpottsoh** [exercism/delphi#280]
@filiptoskovic we can use this issue to communicate.  I have cleaned up your fork of the repo.  Sorry if I strong armed you a little.

I'll be out most of this evening (east coast USA time).  But I'll keep tabs on what you do and check on things tomorrow my time.

**Comment by filiptoskovic** [comment]
Thank you for cleaning. I have added matrix test. Working on Pascal's triangle. Are there some sort of automation or formalism to translate json to DUnitX test project?

**Comment by rpottsoh** [comment]
No there isn't.  Some tracks have developed their own test generators that do what you are thinking, convert the JSON to a test suite for their particular language.  I have not bothered.  So far I have been able to crank out test suites pretty easily.  In the time it would take me to develop a generator I can have several written.  For now, I don't mind the manual exercise.  it helps me to understand the exercise and to arrive at the best test suite for that exercise.

**Comment by filiptoskovic** [comment]
Ok, thank you. Did you have time to look at my tests? Should I make a PR?

**Comment by rpottsoh** [comment]
I have worked on robot-simulator.  Refresh your local copy of the branch (pull).  It is ready for a PR to be made.  There are a lot of commits, that are not really needed.  Before the PR is merged I will squash the commits down to a single commit.

You did an excellent job with this!  Let me know if you have any questions about the changes I have made, if you have any objections, or would like to make further changes.
